tests/components/tessie/test_coordinator.py

```diff
@@ -0,0 +1,73 @@
+"""Test the Tessie sensor platform."""
+from unittest.mock import patch
+
+from homeassistant.components.tessie.const import DOMAIN, TessieStatus
+from homeassistant.core import HomeAssistant
+
+from .common import (
+    ERROR_AUTH,
+    ERROR_CONNECTION,
+    ERROR_TIMEOUT,
+    ERROR_UNKNOWN,
+    TEST_VEHICLE_STATE_ASLEEP,
+    TEST_VEHICLE_STATE_ONLINE,
+    setup_platform,
+)
+
+
+async def test_coordinator(hass: HomeAssistant) -> None:
+    """Tests that the sensors are correct."""
+
+    entry = await setup_platform(hass)
+    coordinator = hass.data[DOMAIN][entry.entry_id][0]
```

Comment:
Please don't access `hass.data` in new tests https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

> Make sure to not interact with any integration details in tests of integrations. Following this pattern will make the tests more robust for changes in the integration.

Instead move time forward with `async_fire_time_changed`

You can patch once (better to use a fixture) and update the return value each time 
with

`mock_get_state.return_value=XXXX`

Reply:
This is now done. Rather than checking the coordinator data, I added checks against a sensor entity, as the coordinator directly influences that sensor.